src/ui/public/query_bar/components/query_bar.tsx

```diff
@@ -92,27 +106,50 @@ export class QueryBarUI extends Component<Props, State> {
       return null;
     }
 
+    let nextQuery = null;
     if (nextProps.query.query !== prevState.query.query) {
-      return {
-        query: {
-          query: toUser(nextProps.query.query),
-          language: nextProps.query.language,
-        },
-        currentProps: nextProps,
+      nextQuery = {
+        query: toUser(nextProps.query.query),
+        language: nextProps.query.language,
       };
     } else if (nextProps.query.language !== prevState.query.language) {
-      return {
-        query: {
-          query: '',
-          language: nextProps.query.language,
-        },
-        currentProps: nextProps,
+      nextQuery = {
+        query: '',
+        language: nextProps.query.language,
       };
     }
 
-    return { currentProps: nextProps };
+    let nextDateRange = null;
+    if (
+      nextProps.from !== get(prevState, 'currentProps.from') ||
+      nextProps.to !== get(prevState, 'currentProps.to')
+    ) {
+      nextDateRange = {
+        from: nextProps.from,
+        to: nextProps.to,
+      };
+    }
+
+    const nextState: any = {
+      currentProps: nextProps,
+    };
+    if (nextQuery) {
+      nextState.query = nextQuery;
+    }
+    if (nextDateRange) {
+      nextState.from = nextDateRange.from;
+      nextState.to = nextDateRange.to;
+    }
+    return nextState;
   }
 
+  // @ts-ignore
```

Comment:
I think this `@ts-ignore` can be removed by making this `public` instead of `private`.

Reply:
Also I'm not totally sure if it's `private` if the `defaultProps` even works. I haven't tried it out though.

Reply:
Change to `public` does not solve the ts lint problem. I could not figure out why it was complaining so just used the ts-ignore.

defaultProps is working. To verify, do not pass `date-range-from` in map.html. then change default props to something like the below. Notice how the time picker displays uses now-17m.  

```
private static defaultProps = {
    showDatePicker: false,
    dateRangeFrom: 'now-17m',
    dateRangeTo: 'now',
  };
```